src/newcharacter.cpp

```diff
@@ -2293,8 +2296,16 @@ tab_direction set_description(WINDOW *w, player *u, const bool allow_reroll, poi
         const std::string action = ctxt.handle_input();
 
         if (action == "NEXT_TAB") {
-            if( OPTIONS["POINT_DISTRIBUTION"] != "freeform" && !points.is_valid() ) {
-                popup(_("Too many points allocated, change some features and try again."));
+            if (OPTIONS["POINT_DISTRIBUTION"] != "freeform" && !points.is_valid() ) {
+                if( points.skill_points_left() < 0 ) {
+                        popup(_("Too many points allocated, change some features and try again."));
+                } else if( points.trait_points_left() < 0 ) {
+                        popup(_("Too many trait points allocated, change some traits or lower some stats and try again."));
+                } else if( points.stat_points_left() < 0 ) {
+                        popup(_("Too many stat points allocated, lower some stats and try again."));
+                } else {
+                        popup(_("Too many points allocated, change some features and try again."));
+                }
```

Comment:
The order of those should probably be inverted.
Currently it downgrades the problem description:
* If you exhaust all your skill+trait points with traits, you get a message about lacking skills. You can't fix it by lowering your skills because you are still above trait limit
* If you exhaust all your stat+trait points with stats, you get a message about lacking trait points, not lacking stat points